zkvm/tests/zkvm.rs

```diff
@@ -79,3 +89,202 @@ fn issue() {
         }
     };
 }
+
+// questions:
+// - do we have to use separate nonce predicates for the two inputs (to ensure uniqueness if they're same qty?)
```

Comment:
My understanding is that we do need the nonce predicates to be unique to ensure uniqueness!

Reply:
if you use at least one `input`, no need for any `nonce`s. This is because inputs are identified by their txid (where their utxos were created), and txid is guaranteed to be unique recursively.

Reply:
`nonce` entry exists only for two reasons: 
1. make issuance possible when no `input`s are available 
2. have a tool to create singleton flavors (assets that can be issued exactly once with some fixed quantity)

Reply:
FYI, in Bitcoin, the "nonce" is implemented by placing block height into the coinbase tx (the "issuance transaction"):

* [BIP30](https://github.com/bitcoin/bips/blob/master/bip-0030.mediawiki) identified two cases of duplicate txids that shadowed coinbase outputs and introduced a rule to reject duplicates.
* [BIP34](https://github.com/bitcoin/bips/blob/master/bip-0034.mediawiki) introduced mandatory block height into the coinbase tx to make sure that such txs are unique w/o relying on explicit checking for duplicate txids (which does not scale).

Reply:
thanks for the explanation, that makes sense! So that is why in the end-to-end unit tests (https://github.com/interstellar/slingshot/issues/78), the "issue w/o nonce" test is also spending from utxo, right?

Reply:
yup!